src/include/daos_srv/evtree.h

```diff
@@ -61,6 +61,48 @@ struct evt_desc {
 	uint8_t				pt_csum[0];
 };
 
+/**
+ * Callbacks and parameters for evtree descriptor
+ *
+ * NB:
+ * - evtree is a standalone algorithm, it should not depend on the rest part
+ *   of VOS, this function table is abstraction of those direct calls to
+ *   VOS/DTX.
+ *
+ * - Most part of this function table is about undo log callbacks, we might
+ *   want to separate those fuctions to a dedicated function table for undo
+ *   log in the future. So both evtree & dbtree can share the same definition
+ *   of undo log.
+ */
+struct evt_desc_cbs {
+	/**
+	 * callback to free bio address, EVtree does not allocate bio address
+	 * so it wouldn't free it as well, user should provide callback to
+	 * free it.
+	 */
+	int		(*dc_bio_free_cb)(struct umem_instance *umm,
+					  struct evt_desc *desc,
+					  daos_size_t nob, void *args);
+	void		 *dc_bio_free_args;
+	/**
+	 * Availability check, it is for data tracked by DTX undo log.
+	 * It is optional, EVTree always treats data extent is available if
+	 * this method is absent.
+	 */
+	int		(*dc_log_status_cb)(struct umem_instance *umm,
+					    struct evt_desc *desc,
+					    int intent, void *args);
+	void		 *dc_log_status_args;
+	/** Add a descriptor to undo log */
```

Comment:
this is an instance where if we had no direct pointers, a callback would be unecessary